Userland/Libraries/LibJS/Runtime/Value.h

```diff
@@ -458,6 +458,12 @@ class Value {
         return static_cast<i32>(m_value.encoded & 0xFFFFFFFF);
     }
 
+    i32 as_i32_integral_number() const
+    {
+        VERIFY(is_int32() || is_finite_number());
+        return is_int32() ? as_i32() : static_cast<i32>(trunc(as_double()));
+    }
```

Comment:
Yep, that is an oversight. My rationale for using integers is that this function is timing-sensitive, and floating point math can take much longer on some systems, especially division. However, the division is just an easy way to put an upper bound on the loop, to avoid abuse/DoS, and I think a better approach to do that is to just implement the upper bound on `N` as recommended by the spec, and implement the handling of negative values manually. Then we also don't have to worry about integer overflow.